src/Mod/Start/App/DisplayedFilesModel.cpp

```diff
@@ -190,16 +190,50 @@ void DisplayedFilesModel::addFile(const QString& filePath)
     if (!qfi.isReadable()) {
         return;
     }
+
+    // Check if FreeCAD can open the file by its extension
     if (!freecadCanOpen(qfi.suffix())) {
         return;
     }
+
+    // Add file information to cache
     _fileInfoCache.emplace_back(getFileInfo(filePath.toStdString()));
+
+    // If the file is a FreeCAD file (.FCStd)
     if (qfi.completeSuffix() == QLatin1String("FCStd")) {
+        // Load thumbnail from FreeCAD file
         auto thumbnail = loadFCStdThumbnail(filePath.toStdString());
         if (!thumbnail.isEmpty()) {
             _imageCache.insert(filePath, thumbnail);
         }
     }
+    else {
+        // If it is not a FreeCAD file, generate thumbnail using F3D
+        QString thumbnailPath = QString(QLatin1String("%1/%2_thumbnail.png"))
+                                    .arg(QLatin1String("/tmp"))

```

Comment:
the path should not be hardcoded, you should be using `Application::getTempPath()` I believe

Reply:
Better use https://doc.qt.io/qt-6/qdir.html#tempPath

Reply:
should we be removing `Application::getTempPath()` if it's better to use the one provided by qt?

Reply:
> should we be removing Application::getTempPath()

No. Application::getTempPath() points to a different directory (can be even customized by the user) than QDir::tempPath() which is not cleaned up by the system after a reboot.

And since the currently created thumbnails of FCStd files are saved to /tmp (under Linux) it makes no sense for me to choose a different directory for f3d thumbnails.